plugin.rb

```diff
@@ -301,68 +314,83 @@ def self.slack_message(post, channel)
 
     def self.absolute(raw)
       url = URI(raw) rescue nil # No icon URL if not valid
+
       if url && url.scheme != 'mailto'
         url.host = Discourse.current_hostname if !(url.host)
         url.scheme = (SiteSetting.force_https ? "https" : "http") if !(url.scheme)
       end
+
       url
     end
 
-    def self.set_filter_by_id(id, channel, filter, channel_id = nil)
+    def self.get_key(id = nil)
+      return "all_categories" unless id.present?
+      "category_#{id}"
+    end
+
+    def self.set_filter_by_id(id, channel, filter, tags = nil, channel_id = nil)
+      id = nil unless id.present?
+      tags = nil unless tags.present?
+
       data = get_store(id)
 
-      update = data.index {|i| i['channel'] === channel || i['channel'] === channel_id }
+      update = data.index {|i| (i['channel'] === channel || i['channel'] === channel_id) && i['tags'] === tags }
 
       if update
         data[update]['filter'] = filter
         data[update]['channel'] = channel # fix old IDs
       else
-        data.push({ channel: channel, filter: filter })
+        data.push({ channel: channel, filter: filter, tags: tags })
       end
 
-      data = data.uniq { |i| i['channel'] }
+      data = data.uniq { |i| i.values_at('channel', 'tags') }
 
-      ::PluginStore.set(PLUGIN_NAME, "category_#{id}", data.uniq)
+      ::PluginStore.set(DiscourseSlack::PLUGIN_NAME, get_key(id), data.uniq)
     end
 
-    def self.delete_filter(id, channel)
+    def self.delete_filter(id, channel, tags)
       data = get_store(id)
       data.delete_if do |i|
         i['channel'] === channel
+        i['tags'] === tags
```

Comment:
This actually broke deleting a filter because it was no longer matching by channels anymore.